src/components/storage/MountPointMapping.jsx

```diff
@@ -264,10 +264,20 @@ const DeviceColumnSelect = ({ deviceData, devices, handleRequestChange, idPrefix
     const size = cockpit.format_bytes(deviceData[device]?.total.v);
     const options = devices.map(device => {
         const deviceName = deviceData[device].name.v;
+
+        const ancestors = getDeviceAncestors(deviceData, device);
+        const parentPartition = [device, ...ancestors].find(ancestor => deviceData[ancestor].type.v === "partition");
```

Comment:
Can't there be a situation where the parent partition is null? Because that could cause an error

Reply:
Do you mean that: deviceData[ancestor] is undefined?

Well ancestors should be always an array - sometimes an empty one.

Reply:
Because in the following code we feel that we always get the parentPartition, so if there can't be some edge case in which this doesn't work

Reply:
Hm I see what you mean now.

I really cant imagine how we can have a storage layout where there is not a partition in the device tree.

however you 're right - we shall be better be safe - will adjust.

Reply:
Did the safe access change.